linked_list/linked_list.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import print_function  # For Python 2
+
+
+class node(object):
+    def __init__(self, data=None, next_=None):
+        # Initialises Node
+        self.data = data
+        self.next = next_
+
+    def __str__(self):
+        return str(self.data)
+
+
+class linked_list(object):
+    def __init__(self):
+        # Initialises Linked List.
+        self.head = None
+        self.size = 0
+
+    def add_front(self, data):
+        # Inserts element at first position.
+        n = node(data)
+        n.next = self.head
+        self.head = n
+        self.size += 1
+
+    def add_last(self, data):
+        # Inserts element at last position.
+        if not self.head:
+            n = node(data)
+            self.head = n
+            self.size = 1
+            return
+        else:
+            n = self.head
+            while n.next is not None:
+                n = n.next
+            new_node = node(data)
+            n.next = new_node
+            self.size += 1
+            return
+
+    def add(self, data, index):
+        # Inserts data at index.
+        if index < 0 or index > self.size:
+            raise IndexError
+        temp = self.head
+        if index is 0:
+            add_front(data)
+            return
+
+        else:
+            for _ in range(index - 1):
+                temp = temp.next
+            n = node(data)
+            n.next = temp.next
+            temp.next = n
+            self.size += 1
+            return
+
+    def remove_front(self):
+        # Deletes First Element
+        n = self.head
+        if self.is_empty():
+            raise ValueError
+        removed = n.data
+        n = n.next
+        self.size -= 1
+        return removed
+
+    def remove_last(self):
```

Comment:
Both remove_front and remove_last should work using a call to `remove` method. 
No need to duplicate code.